src/metabase/api/card.clj

```diff
@@ -775,26 +470,26 @@ saved later when it is ready."
                check-allowed-to-change-embedding]]
       (f card-before-update card-updates))
     ;; make sure we have the correct `result_metadata`
-    (let [result-metadata-chan  (result-metadata-async {:original-query    (:dataset_query card-before-update)
-                                                        :query             dataset_query
-                                                        :metadata          result_metadata
-                                                        :original-metadata (:result_metadata card-before-update)
-                                                        :dataset?          (if (some? dataset)
-                                                                             dataset
-                                                                             (:dataset card-before-update))})
+    (let [result-metadata-chan  (card/result-metadata-async {:original-query    (:dataset_query card-before-update)
```

Comment:
This timeout stuff should probably be extracted into a function in the `metabase.models.card` namespace too, but I didn't want to do too much at once